### ISSUE TITLE
Bulk Users import fails

### ISSUE THREAD

**Issue by tvsjr** [gophish/gophish#1053]
What version of Gophish are you using?: 
0.5.0, latest version pulled from GitHub and compiled (to avoid the failure to start issue present in the currently available compiled version)

Brief description of the issue: 
Bulk imports from a CSV file to a Group fail.

What are you expecting to see happen? :
Using a CSV file that imported properly in 0.4.0, I expect to see the users import into a new group.

What are you seeing happen?
No import occurs. Watching the process with Fiddler, GoPhish responds with "API Key not set", specifically:
{
  "message": "API Key not set",
  "success": false,
  "data": null
}

I can manually enter new users, typing each one in... but manually adding 1K users isn't a great option!

Please provide as many steps as you can to reproduce the problem:
Log into the administrative interface.
Go to Users and Groups.
Create a New Group.
Name the group.
Select Bulk Import Users.
Choose a known-good CSV file.
Select OK.
Observe that no users are imported.

**Comment by jordan-wright** [comment]
Hmmm... this looks like it's probably a regression somewhere. I'll see if I can reproduce the issue and get it fixed pronto.

Thanks for the heads up! Sorry for the inconvenience.

**Comment by tvsjr** [comment]
I ended up importing these manually by directly adding them to the MySQL database. Now, when attempting to launch a campaign, I receive:
"Error 1054: Unknown column 'reported' in 'field list'"

I did build from source, and this seems to be a similar issue to the one reported in issue 1035.

If your comments in 1035 are accurate, that perhaps the database didn't upgrade correctly, then perhaps this is the root cause of both issues.

I'm going to dig into the issue a bit to see if I can resolve it (or, perhaps, a clean install might resolve the issue long enough to get my current campaign transmitted).

**Comment by tvsjr** [comment]
More narrowing down...

I built a totally new install from source, new database, everything. Fired it up, generated a simple campaign, one target entered manually, etc. Go to send the campaign, and I get the same Error 1054 reported above. This is with the MySQL backend.

So, I switched it over to SQLite... the error is different, but similar: "table results has no column named reported".

I went through the migrations file and found the 0.5.1 migration that adds the "reported" column. I went back to the original MySQL configuration, executed that ALTER TABLE statement and, huzzah!, the campaign works. But, the bulk import from a known-good CSV file still fails.

tl;dr - it appears the 0.5.1 SQL migration isn't being executed, thus causing the secondary issue. However, this did *not* fix the initial issue... not being able to do bulk imports. So, there's probably two bugs here, potentially related... the bulk import issue, and the migration script not being run. If you want me to file a second ticket, let me know and I will.

**Comment by jordan-wright** [comment]
Just curious, when building from source, did you also make sure that the new Javascript files are being used? Part of 2131c17c33a0af757c80d94379baaca147de9931 involved updating all API endpoints to require an API key. This meant that I had to update some JS files to make sure they were sending the API when making the call.

It's possible that the new users.min.js isn't being used by your Gophish installation, which would cause the request to be sent without the API key.

Let me know if this helps! Otherwise, we can continue troubleshooting.

**Comment by jordan-wright** [comment]
Also, I can't seem to replicate the "no 'reported' column" error. This is the output of building Gophish from source and running it on a new machine:

```
# ./gophish
 2018/05/03 23:43:28 worker.go:26: Background Worker Started Successfully - Waiting for Campaigns
goose: migrating db environment 'production', current version: 0, target: 20180223101813
OK    20160118194630_init.sql
OK    20160131153104_0.1.2_add_event_details.sql
OK    20160211211220_0.1.2_add_ignore_cert_errors.sql
OK    20160217211342_0.1.2_create_from_col_results.sql
OK    20160225173824_0.1.2_capture_credentials.sql
OK    20160227180335_0.1.2_store-smtp-settings.sql
OK    20160317214457_0.2_redirect_url.sql
OK    20160605210903_0.2_campaign_scheduling.sql
OK    20170104220731_0.2_result_statuses.sql
OK    20170219122503_0.2.1_email_headers.sql
OK    20170827141312_0.4_utc_dates.sql
OK    20171027213457_0.4.1_maillogs.sql
OK    20171208201932_0.4.1_next_send_date.sql
OK    20180223101813_0.5.1_user_reporting.sql
```

As you can see at the bottom, the migration is performed. Just curious, do you see that file 20180223101813_0.5.1_user_reporting.sql in your db/db_sqlite3/migrations/ folder?

**Comment by SpaWn2KiLl** [comment]
Solved this issue by manually inserting api_key to the URL in the JQuery Fileupload function inside the users.js file

`$("#csvupload").fileupload({
        dataType: "json",
        add: function(e, data) {
            $("#modal\\.flashes").empty()
            var acceptFileTypes = /(csv|txt)$/i;
            var filename = data.originalFiles[0]['name']
            if (filename && !acceptFileTypes.test(filename.split(".").pop())) {
                modalError("Unsupported file extension (use .csv or .txt)")
                return false;
            }
            **data.url = '/api/import/group?api_key=' + user.api_key;**
            data.submit();
        },
        done: function(e, data) {
            $.each(data.result, function(i, record) {
                addTarget(
                    record.first_name,
                    record.last_name,
                    record.email,
                    record.position);
            });
            targets.DataTable().draw();
        }
    })`

**Comment by jordan-wright** [comment]
I believe both of the issues mentioned in here should be resolved by upgrading to v0.6.0, which includes both the migration to add the reported status as well as the JS update to send the API key for uploading users.

With this being the case, I'm going to go ahead and close this one out. Thanks again for getting in touch!
